pkg/lifecycle/service.go

```diff
@@ -431,6 +431,7 @@ func (s *Service) buildRunnablePipeline(
 ) (*runnablePipeline, error) {
 	nodes, err := s.buildNodes(ctx, pl)
 	if err != nil {
+		pl.SetStatus(pipeline.StatusDegraded)
```

Comment:
I don't understand why we would change the status, if we didn't even start the pipeline yet? 🤔 Is there an issue connected to this PR that describes what problem we're trying to solve?

Reply:
was initially linked to issue #1852 but it was fixed from pipeline recovery epic. Will close this now.